File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution(object):
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        m = len(matrix[0])

        row = [False for _ in range(n)]
        col = [False for _ in range(m)]

        for i in range(n):
            for j in range(m):
                if matrix[i][j] == 0:
                    row[i] = True
                    col[j] = True
        
        for i in range(n):
            for j in range(m):
                if row[i] or col[j]:
                    matrix[i][j] = 0

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cs
public class Solution {
    public void SetZeroes(int[][] matrix) {
        var n = matrix.Length;
        var m = matrix[0].Length;

        bool[] row = new bool[n];
        bool[] col = new bool[m];
        for (var i = 0; i < n; i++) {
            for (var j = 0; j < m; j++) {
                if (matrix[i][j] == 0) {
                    row[i] = true;
                    col[j] = true;
                }
            }
        }

        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                if (row[i] || col[j])
                    matrix[i][j] = 0;
            }
        }
    }
}